### ISSUE TITLE
@JsonUnwrapped incorrectry rewrites field defined by @JsonCreator

### ISSUE THREAD

**Issue by vananiev** [FasterXML/jackson-databind#3127]
For version 2.11.4 `BeanDeserializer.java:835` line
```java
return _unwrappedPropertyHandler.processUnwrapped(p, ctxt, bean, tokens);
```
rewrites `@JsonUnwrapped` annotated field value by object containing nulls. Originally correct field value defined by `@JsonCreator` constructor.

**Comment by cowtowncoder** [comment]
I would need a test case to show exactly what you mean -- not doubting there is a problem, just difficult to visualize without class definition(s) and JSON input.

**Comment by vananiev** [comment]
Try this class
```java
@Data
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class DnsRecord {

    @JsonUnwrapped // <- replaces deserialized 'key' field value by method BeanDeserializer.java:835
    private final DnsKey key;

//    with this code works (@JsonUnwrapped for getter, ignore field for setter)
//
//    @JsonIgnore
//    private final DnsKey key;
//
//    @JsonUnwrapped
//    public DnsKey getKey() {
//        return key;
//    }

    @JsonProperty("info")
    private final Set<Info> info;

    @JsonCreator
    private static DnsRecord of(@JsonProperty("ip") String serverIp,
                                @JsonProperty("port") Integer serverPort,
                                @JsonProperty("info") Set<Info> info) {
        var key = DnsKey.builder()
                .serverIp(serverIp)
                .serverPort(serverPort)
                .build();
        return DnsRecord.builder()
                .key(key)
                .info(info)
                .build();
    }

    @Getter
    @Jacksonized
    @Builder
    private static class DnsKey {

        @JsonProperty("ip")
        private final String serverIp;

        @JsonProperty("port")
        private final Integer serverPort;
    }

    @Builder
    @Jacksonized
    @EqualsAndHashCode
    public static class Info {

        @JsonProperty("key")
        private final String key;

        @JsonProperty("value")
        private final String value;
    }
}
```

**Comment by cowtowncoder** [comment]
Cannot reproduce due to reproduction using Lombok (or similar framework), but expecting this would be fixed by #1467 (included for upcoming 2.19.0). Closing; may be re-opened with stand-alone reproduction showing the issue against `2.19` branch.
